operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -158,26 +185,50 @@ public static File createFileStore(String prefix, String suffix, byte[] bytes) {
      * Decode binary item from Kubernetes Secret from base64 into byte array
      *
      * @param secret    Kubernetes Secret
-     * @param key       Key which should be retrieved and decoded
+     * @param field     Field which should be retrieved and decoded
      * @return          Decoded bytes
      */
-    public static byte[] decodeFromSecret(Secret secret, String key) {
-        return Base64.getDecoder().decode(secret.getData().get(key));
+    public static byte[] decodeBase64FieldFromSecret(Secret secret, String field) {
+        return Optional.ofNullable(secret)
+                .map(Secret::getData)
+                .map(data -> data.get(field))
+                .map(value -> Base64.getDecoder().decode(value))
+                .orElseThrow(() -> {
+                    String name = Optional.ofNullable(secret)
+                            .map(Secret::getMetadata)
+                            .map(ObjectMeta::getName)
+                            .orElse("unknown");
+                    String namespace = Optional.ofNullable(secret)
+                            .map(Secret::getMetadata)
+                            .map(ObjectMeta::getNamespace)
+                            .orElse("unknown");
+                    return Util.missingDataInSecretException(namespace, name, field);
+                });
     }
 
     /**
-     * Decode the binary item in a Kubernetes Secret, which holds a private key in PEM format, from base64 to a byte array.
-     * Before decoding it into byte array, it removes the PEM header and footer.
+     * Decode binary item from Kubernetes Secret from base64 into String
+     *
      * @param secret    Kubernetes Secret
-     * @param key       Key which should be retrieved and decoded
-     * @return          Decoded bytes
+     * @param field     Field which should be retrieved and decoded
+     * @return          Decoded String
+     */
+    public static String asciiFieldFromSecret(Secret secret, String field) {
+        return fromAsciiBytes(decodeBase64FieldFromSecret(secret, field));
+    }
+
+    /**
+     * Decode the private key in PEM format, from base64 to a byte array.
+     * Before decoding it into byte array, it removes the PEM header and footer.
+     * @param privateKey        Key which should be decoded
+     * @return                  Decoded bytes
      */
-    public static byte[] decodePemPrivateKeyFromSecret(Secret secret, String key) {
-        String privateKey = new String(decodeFromSecret(secret, key), StandardCharsets.UTF_8)
+    public static byte[] decodePemPrivateKey(String privateKey) {
+        String decodedPrivateKey = privateKey
                 .replace("-----BEGIN PRIVATE KEY-----", "")
```

Comment:
I agree that we could have better validation here. However I think we should include it in a follow up PR where we better align the creation of keystores and truststores across the project, since I don't think this should stay in Util ultimately.